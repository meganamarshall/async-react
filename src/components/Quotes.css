.Quotes {
  border: 1px solid black;
  text-decoration: none;
  display: flex;
  flex-wrap: wrap;

}
