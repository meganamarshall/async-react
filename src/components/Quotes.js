import React from 'react';
import PropTypes from 'prop-types';
import Quote from './Quote';
import styles from './Quotes.css';

function Quotes({ quotes }) {
  const quoteList = quotes.map(quote => (
    <li key={quote.quote}>
      <Quote quote={quote} />
    </li>
  )
  );

  return (
    <ul className={styles.Quotes}>
      {quoteList}
    </ul>
  );
}

Quotes.propTypes = {
  quotes: PropTypes.array.isRequired
};

export default Quotes;
