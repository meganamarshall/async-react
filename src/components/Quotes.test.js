import React from 'react';
import { shallow } from 'enzyme';
import Quotes from './Quotes';

describe('quotes component', () => {
  it('renders the quotes', () => {
    const quotes = [{
      character: 'Megan',
      quote: 'Hi',
      image: 'imagepath.png'
    },
    {
      character: 'Bob',
      quote: 'Bye',
      image: 'byepic.jpg'
    }];
    const wrapper = shallow(<Quotes quotes={quotes} />);
    expect(wrapper).toMatchSnapshot();
  });
});
