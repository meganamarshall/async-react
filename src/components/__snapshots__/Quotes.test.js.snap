// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quotes component renders the quotes 1`] = `
<ul>
  <li
    key="Hi"
  >
    <Quote
      quote={
        Object {
          "character": "Megan",
          "image": "imagepath.png",
          "quote": "Hi",
        }
      }
    />
  </li>
  <li
    key="Bye"
  >
    <Quote
      quote={
        Object {
          "character": "Bob",
          "image": "byepic.jpg",
          "quote": "Bye",
        }
      }
    />
  </li>
</ul>
`;
